Surplus/executed popup

nenadV91 opened the pull request:
# Summary

Part of the [Surplus tasks](https://www.notion.so/cownation/Show-Me-the-Money-hackathon-0a1b1b3e6a1b4d4eb9a46f96136a1865)

This PR
- updates the message in executed order popup, as requested `Traded <amount TOKEN> for a total of <traded amount TOKEN>` and some styling
- adds surplus amount to the popup

![Screenshot 2023-04-11 at 16 12 09](https://user-images.githubusercontent.com/34926005/231196329-981558ec-1cc9-4910-969e-50329b021d71.png)

![Screenshot 2023-04-11 at 16 34 47](https://user-images.githubusercontent.com/34926005/231196889-eeb9d3ff-b32e-4dc3-b831-4b1b82134af3.png)

elena-zh reviewed (changes requested):
Hey @nenadV91 , cool!
Some issues from my side: 
1. Total amount still shows at least, not the final amount 
![total tradeda mount](https://user-images.githubusercontent.com/70885163/231214084-2875d885-cde1-4306-9672-09e2d672c8bd.png)
2, Pop-up for a filled limit order is no longer displayed in your PR
3. To clarify: there is no icon on the pop-up, and the checkmark icon still looks different. Will it be addressed in another PR? 
![image](https://user-images.githubusercontent.com/70885163/231214729-41ca0129-a2c2-4d1c-871d-2545cb29d9d3.png)
Thanks!

shoom3301 commented on src/custom/state/orders/helpers.tsx:
```diff
@@ -49,11 +55,22 @@ type MetaPopupContent = GPPopupContent<OrderTxTypes.METATXN>
-    : typeof summary === 'string'
-    ? `${summary} ${descriptor || status || ''}`
-    : summary
+  // If there isn't summary, return generalized summary
```
Awesome refactoring! Now it much understandable

shoom3301 commented on src/custom/state/orders/helpers.tsx:
```diff
@@ -122,3 +139,77 @@ export function setPopupData(
+  const inputT = order.inputToken
+  const outputT = order.outputToken
+
+  const inputToken = new Token(inputT.chainId, inputT.address, inputT.decimals, inputT.symbol, inputT.name)
```
Why do we need this token mapping? Can we just use `order.inputToken`?

nenadV91 commented on src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:
```diff
@@ -1,83 +1,19 @@
 
-// TODO: using .toNumber() we potentially lose accuracy
-// TODO: must be refactored with replacing bignumber.js by @ethersproject/bignumber
-function legacyBigNumberToCurrencyAmount(currency: Token, value: BigNumber | undefined): CurrencyAmount<Token> {
```
Moved all of this code to a separate util function `src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts` because we also need the same values in execution popup, so we can reuse it.

anxolin replied:
Great to see this code moved to an util.

Why this util is inside `limitOrders` module? isn't generic for anything that has to do with orders?

nenadV91 commented on src/cow-react/modules/limitOrders/pure/ReceiptModal/index.tsx:
```diff
@@ -114,7 +113,7 @@ export function ReceiptModal({
 
             <styledEl.Field>
               <FieldLabel label="Filled" tooltip={tooltips.FILLED} />
-              <FilledField order={order} sellAmount={sellAmount} buyAmount={buyAmount} />
```
Don't need this anymore

nenadV91 commented on src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts:
```diff
@@ -0,0 +1,78 @@
+  const sellAmountCurrency = CurrencyAmount.fromRawAmount(inputToken, sellAmount.toString())
+  const buyAmountCurrency = CurrencyAmount.fromRawAmount(outputToken, buyAmount.toString())
+
+  // TODO: set types, move calculations logic to a function
```
I guess we did this now

anxolin commented on src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx:
```diff
@@ -1,83 +1,19 @@
+  const { filledPercentage, formattedPercentage, fullyFilled } = order
+  const { action, mainAmount, formattedFilledAmount, formattedSwappedAmount } = getFilledAmounts(order)
 
   const touched = !!filledPercentage?.gt(0)
```
does this need the !!? is not `gt` returning a boolean already?

nenadV91 commented:
@elena-zh 
1. Should be fixed now
2. Works fine for me, maybe new changes fixed it
3. I think this is fine for now, if we have more time we can update the UI more

anxolin commented on src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts:
```diff
@@ -0,0 +1,78 @@
+import JSBI from 'jsbi'
+
+// TODO: using .toNumber() we potentially lose accuracy
+// TODO: must be refactored with replacing bignumber.js by @ethersproject/bignumber
```
If we do migrations to etherjs v6, we should use native ES6 bignumber

anxolin replied:
maybe for now we can just change the TODO

nenadV91 replied:
Updated todo

anxolin commented on src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts:
```diff
@@ -0,0 +1,78 @@
+import { ParsedOrder } from '@cow/modules/limitOrders/containers/OrdersWidget/hooks/useLimitOrdersList'
```
I see why stays in the module of limit orders. This is specific to this ParsedOrders which is for now only in limit orders module

I think in the future we might want to move some of these things to some orders module that is independent. We might have swap, limit orders, twap, classical trading UI, etc and in principle all this things are the same for all

anxolin commented on src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts:
```diff
@@ -0,0 +1,78 @@
+  let mainAmount: CurrencyAmount<Token>
+  let swappedToken: Token
+  let swappedAmount: JSBI | undefined
+  let action: string
```
I know someone that won't be happy with so many `let`

nenadV91 replied:
Honestly in this case I prefer this instead of having a ternary operator for each variable.

anxolin commented on src/custom/state/orders/helpers.tsx:
```diff
@@ -122,3 +139,85 @@ export function setPopupData(
+  margin-right: 5px;
+`
+
+export function getExecutedSummary(order: ParsedOrder): JSX.Element | string {
```
should this be a dump component?  why is in the `state`?

anxolin reviewed (commented):
Nenad, would it be good to have a cosmos with all the "cases" you want to cover? 

This will make the reviewers to give you a better feedback after you present all representative cases.

MindyCoW commented:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/104489816/231702668-5de632c6-af11-4003-a910-738ab31f746d.png">

As Elena said, we should show the token amount and the USD estimation. No need to show the percentage IMO.

nenadV91 commented:
@elena-zh @MindyCoW I've removed the percentage from surplus. 
USD estimation is good to have and will not be a part of this PR.

nenadV91 commented:
@anxolin 

> Nenad, would it be good to have a cosmos with all the "cases" you want to cover?
> This will make the reviewers to give you a better feedback after you present all representative cases.

For now I don't think we need this since we will only change executed popup message.
But its something we can do in the future if we want to.

elena-zh reviewed (approved):
I believe that the pop-up styles will be updated in a separate PR. 
The current changes LGTM

nenadV91 commented:
@anxolin @shoom3301 I've addressed most of the comments.
Will merge these surplus PR's since Elena approved it and I'm not here till Tuesday.
But add any other comments you have and I will address them in a separate PR
